src/widget/wpixmapstore.cpp

```diff
@@ -14,32 +14,30 @@ PaintablePointer WPixmapStore::getPaintable(const PixmapSource& source,
     if (source.isEmpty()) {
         return PaintablePointer();
     }
-    QString key = source.getId() + QString::number(mode) + QString::number(scaleFactor);
+    QString key = QStringLiteral("%1%2%3").arg(source.getId(),
+            QString::number(mode),
+            QString::number(scaleFactor));
 
     // See if we have a cached value for the pixmap.
-    PaintablePointer pPaintable = m_paintableCache.value(
-            key,
-            PaintablePointer());
-    if (pPaintable) {
-        return pPaintable;
+    auto it = m_paintableCache.find(key);
+    if (it != m_paintableCache.end()) {
+        return it.value();
     }
 
-    pPaintable = PaintablePointer(new Paintable(source, mode, scaleFactor));
-
+    PaintablePointer pPaintable(new Paintable(source, mode, scaleFactor));
```

Comment:
this is a `QSharedPointer<Paintable>`, right?
```suggestion
    auto pPaintable = PaintablePointer::create(source, mode, scaleFactor);
```